ledger response deserialization failure

Hello Team,

Got another issue in parsing response from the `ledger` call. Seems to me like an edge case that was missed out. Seems to be failing for `73151744` and `73151488` with the same error.

The calling code is as follows:
```java
client.ledger(LedgerRequestParams.builder()
            .ledgerIndex(LedgerIndex.of(UnsignedInteger.valueOf(index)))
            .transactions(true)
            .build());
```

Rippled server is https://s2-reporting.ripple.com:51234

Here is the log for one of them:
```
could not get ledger info for ledger, index=73151744, blockchain/network=XRP/LIVENET, error=com.fasterxml.jackson.databind.JsonMappingException: No matching TransactionType enum value for String value UNLModify (through reference chain: org.xrpl.xrpl4j.model.client.ledger.ImmutableLedgerResult$Json["ledger"]->org.xrpl.xrpl4j.model.ledger.ImmutableLedgerHeader$Json["transactions"]->java.util.ArrayList[46])
````

Stack trace:
```
org.xrpl.xrpl4j.client.JsonRpcClientErrorException: com.fasterxml.jackson.databind.JsonMappingException: No matching TransactionType enum value for String value UNLModify (through reference chain: org.xrpl.xrpl4j.model.client.ledger.ImmutableLedgerResult$Json["ledger"]->org.xrpl.xrpl4j.model.ledger.ImmutableLedgerHeader$Json["transactions"]->java.util.ArrayList[19])
	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:120) ~[xrpl4j-client-2.3.0.jar:na]
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:731) ~[na:na]
	at feign.DefaultMethodHandler.invoke(DefaultMethodHandler.java:141) ~[feign-core-11.6.jar:na]
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:100) ~[feign-core-11.6.jar:na]
	at jdk.proxy2/jdk.proxy2.$Proxy142.send(Unknown Source) ~[na:na]
	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:93) ~[xrpl4j-client-2.3.0.jar:na]
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:731) ~[na:na]
	at feign.DefaultMethodHandler.invoke(DefaultMethodHandler.java:141) ~[feign-core-11.6.jar:na]
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:100) ~[feign-core-11.6.jar:na]
	at jdk.proxy2/jdk.proxy2.$Proxy142.send(Unknown Source) ~[na:na]
	at org.xrpl.xrpl4j.client.XrplClient.ledger(XrplClient.java:431) ~[xrpl4j-client-2.3.0.jar:na]
	at co.pal.de.connector.xrp.listeners.XrpListener.handle(ValidatedXrpLedgerListener.java:81) ~[classes/:na]
	at co.pal.de.connector.xrp.listeners.XrpListener$$FastClassBySpringCGLIB$$8489f86f.invoke(<generated>) ~[classes/:na]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89) ~[spring-aop-5.3.18.jar:5.3.18]
	at io.micrometer.core.aop.CountedAspect.interceptAndRecord(CountedAspect.java:199) ~[micrometer-core-1.8.4.jar:1.8.4]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:577) ~[na:na]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89) ~[spring-aop-5.3.18.jar:5.3.18]
	at io.micrometer.core.aop.TimedAspect.processWithTimer(TimedAspect.java:192) ~[micrometer-core-1.8.4.jar:1.8.4]
	at io.micrometer.core.aop.TimedAspect.timedMethod(TimedAspect.java:170) ~[micrometer-core-1.8.4.jar:1.8.4]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:577) ~[na:na]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar:5.3.18]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698) ~[spring-aop-5.3.18.jar:5.3.18]
	at co.pal.de.connector.xrp.listeners.XrpListener$$EnhancerBySpringCGLIB$$35b56bb7.handle(<generated>) ~[classes/:na]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:577) ~[na:na]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:169) ~[spring-messaging-5.3.18.jar:5.3.18]
	at org.springframework.messaging.handler.invocation.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:119) ~[spring-messaging-5.3.18.jar:5.3.18]
	at org.springframework.amqp.rabbit.listener.adapter.HandlerAdapter.invoke(HandlerAdapter.java:75) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.adapter.MessagingMessageListenerAdapter.invokeHandler(MessagingMessageListenerAdapter.java:261) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.adapter.MessagingMessageListenerAdapter.invokeHandlerAndProcessResult(MessagingMessageListenerAdapter.java:207) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.adapter.MessagingMessageListenerAdapter.onMessage(MessagingMessageListenerAdapter.java:146) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doInvokeListener(AbstractMessageListenerContainer.java:1665) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.actualInvokeListener(AbstractMessageListenerContainer.java:1584) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.invokeListener(AbstractMessageListenerContainer.java:1572) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.doExecuteListener(AbstractMessageListenerContainer.java:1563) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer.executeListener(AbstractMessageListenerContainer.java:1507) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.doReceiveAndExecute(SimpleMessageListenerContainer.java:967) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.receiveAndExecute(SimpleMessageListenerContainer.java:914) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer.access$1600(SimpleMessageListenerContainer.java:83) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.mainLoop(SimpleMessageListenerContainer.java:1291) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer$AsyncMessageProcessingConsumer.run(SimpleMessageListenerContainer.java:1197) ~[spring-rabbit-2.4.3.jar:2.4.3]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No matching TransactionType enum value for String value UNLModify (through reference chain: org.xrpl.xrpl4j.model.client.ledger.ImmutableLedgerResult$Json["ledger"]->org.xrpl.xrpl4j.model.ledger.ImmutableLedgerHeader$Json["transactions"]->java.util.ArrayList[19])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:392) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:363) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:371) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:214) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:186) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1398) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:214) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:186) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1398) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:118) ~[xrpl4j-client-2.3.0.jar:na]
	... 59 common frames omitted
Caused by: java.lang.IllegalArgumentException: No matching TransactionType enum value for String value UNLModify
	at org.xrpl.xrpl4j.model.transactions.TransactionType.forValue(TransactionType.java:125) ~[xrpl4j-model-2.3.0.jar:na]
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionDeserializer.deserialize(TransactionDeserializer.java:31) ~[xrpl4j-model-2.3.0.jar:na]
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionDeserializer.deserialize(TransactionDeserializer.java:17) ~[xrpl4j-model-2.3.0.jar:na]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4388) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4344) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionResultDeserializer.deserialize(TransactionResultDeserializer.java:45) ~[xrpl4j-model-2.3.0.jar:na]
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionResultDeserializer.deserialize(TransactionResultDeserializer.java:29) ~[xrpl4j-model-2.3.0.jar:na]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:355) ~[jackson-databind-2.13.2.2.jar:2.13.2.2]
	... 79 common frames omitted
```

I was just about to cut v2.4 release, but it looks like this one is still an issue in `main` (though for different reasons). Basically, XRPL is returning an empty `Account` field for UNLModify transactions, which triggers the stack trace at the end of this comment.

TBH I'm not entirely sure how to process on this one. Some options:

1. Custom deserializer that uses a dummy address in the `Account` field (blah!) so we don't have to make this field `Optional` for _every_ `transaction`
2. Don't allow `UnlModify` to extend `Transaction` and only allow [these fields](https://xrpl.org/unlmodify.html#example-unlmodify-json)?
3. Change the JSON contract for `Transaction` to make `account()` be `@Nullable` and violate our "[null is a bug](https://github.com/google/guava/wiki/UsingAndAvoidingNullExplained)" rule?
4. Change the contract for `Transaction.account()` to make it return an `Optional`?

Overall I like none of these options, but maybe option-2 is tenable?

### Java Code
```java
    LedgerResult ledger = xrplClient
      .ledger(LedgerRequestParams.builder()
        .ledgerIndex(LedgerIndex.of(UnsignedInteger.valueOf(73151744)))
        .transactions(true)
        .build());
```

### JSON Response Snippet

```json
{
          "Account": "",
          "Fee": "0",
          "LedgerSequence": 73151488,
          "Sequence": 0,
          "SigningPubKey": "",
          "TransactionType": "UNLModify",
          "UNLModifyDisabling": 1,
          "UNLModifyValidator": "ED583ECD06C3B7369980E65C78C440A529300F557ED81256283F7DD5AA3513A334",
          "hash": "04EE2615FD57DA2AF3ED2C32DAC6F9557596E7A70A9A7EDE32BF9E9D4FA6A39D",
          "metaData": {
            "AffectedNodes": [
              {
                "CreatedNode": {
                  "LedgerEntryType": "NegativeUNL",
                  "LedgerIndex": "2E8A59AA9D3B5B186B0B9E0F62E6C02587CA74A4D778938E957B6357D364B244",
                  "NewFields": {
                    "ValidatorToDisable": "ED583ECD06C3B7369980E65C78C440A529300F557ED81256283F7DD5AA3513A334"
                  }
                }
              }
            ],
            "TransactionIndex": 19,
            "TransactionResult": "tesSUCCESS"
          }
        },
```

### Stack Trace
```
org.xrpl.xrpl4j.client.JsonRpcClientErrorException: com.fasterxml.jackson.databind.JsonMappingException: Invalid Address: Bad Prefix (through reference chain: org.xrpl.xrpl4j.model.transactions.ImmutableUnlModify$Json["Account"]) (through reference chain: org.xrpl.xrpl4j.model.client.ledger.ImmutableLedgerResult$Json["ledger"]->org.xrpl.xrpl4j.model.ledger.ImmutableLedgerHeader$Json["transactions"]->java.util.ArrayList[2])

	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:142)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at feign.DefaultMethodHandler.invoke(DefaultMethodHandler.java:141)
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:100)
	at jdk.proxy2/jdk.proxy2.$Proxy12.send(Unknown Source)
	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:114)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at feign.DefaultMethodHandler.invoke(DefaultMethodHandler.java:141)
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:100)
	at jdk.proxy2/jdk.proxy2.$Proxy12.send(Unknown Source)
	at org.xrpl.xrpl4j.client.XrplClient.ledger(XrplClient.java:656)
	at org.xrpl.xrpl4j.tests.AccountSetIT.foo(AccountSetIT.java:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Invalid Address: Bad Prefix (through reference chain: org.xrpl.xrpl4j.model.transactions.ImmutableUnlModify$Json["Account"]) (through reference chain: org.xrpl.xrpl4j.model.client.ledger.ImmutableLedgerResult$Json["ledger"]->org.xrpl.xrpl4j.model.ledger.ImmutableLedgerHeader$Json["transactions"]->java.util.ArrayList[2])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:392)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:363)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:371)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:186)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1398)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:186)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1398)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629)
	at org.xrpl.xrpl4j.client.JsonRpcClient.send(JsonRpcClient.java:140)
	... 78 more
Caused by: java.lang.IllegalArgumentException: Invalid Address: Bad Prefix (through reference chain: org.xrpl.xrpl4j.model.transactions.ImmutableUnlModify$Json["Account"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4393)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4344)
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionResultDeserializer.deserialize(TransactionResultDeserializer.java:65)
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionResultDeserializer.deserialize(TransactionResultDeserializer.java:49)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:355)
	... 98 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Invalid Address: Bad Prefix (through reference chain: org.xrpl.xrpl4j.model.transactions.ImmutableUnlModify$Json["Account"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:392)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1821)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:315)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:214)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:186)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1398)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:351)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:184)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4650)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2831)
	at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:3295)
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionDeserializer.deserialize(TransactionDeserializer.java:52)
	at org.xrpl.xrpl4j.model.jackson.modules.TransactionDeserializer.deserialize(TransactionDeserializer.java:37)
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4388)
	... 102 more
Caused by: java.lang.IllegalArgumentException: Invalid Address: Bad Prefix
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:142)
	at org.xrpl.xrpl4j.model.transactions.Wrappers$_Address.validateAddress(Wrappers.java:64)
	at org.xrpl.xrpl4j.model.transactions.Address.validateAddress(Address.java:22)
	at org.xrpl.xrpl4j.model.transactions.Address.validate(Address.java:105)
	at org.xrpl.xrpl4j.model.transactions.Address.of(Address.java:101)
	at org.xrpl.xrpl4j.model.jackson.modules.AddressDeserializer.deserialize(AddressDeserializer.java:44)
	at org.xrpl.xrpl4j.model.jackson.modules.AddressDeserializer.deserialize(AddressDeserializer.java:33)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:313)
	... 114 more
```

--- a/xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/LedgerResultIT.java
+++ b/xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/LedgerResultIT.java
@@ -9,9 +9,9 @@ package org.xrpl.xrpl4j.tests;
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
+ *
  *      http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
@@ -30,13 +30,19 @@ import org.xrpl.xrpl4j.model.client.common.LedgerSpecifier;
 import org.xrpl.xrpl4j.model.client.ledger.LedgerRequestParams;
 import org.xrpl.xrpl4j.model.client.ledger.LedgerResult;
 import org.xrpl.xrpl4j.model.client.transactions.TransactionResult;
+import org.xrpl.xrpl4j.model.transactions.Transaction;
+import org.xrpl.xrpl4j.model.transactions.UnlModify;
 import org.xrpl.xrpl4j.tests.environment.MainnetEnvironment;
+
+import java.util.Optional;
 
 /**
  * These tests ensure {@link LedgerResult}s can be constructed from all of the different JSON responses
  * rippled sends back.
  */
 public class LedgerResultIT extends AbstractIT {
+
+  final XrplClient mainnetClient = new MainnetEnvironment().getXrplClient();
 
   @Test
   void getValidatedLedgerResult() throws JsonRpcClientErrorException {
@@ -95,8 +101,6 @@ public class LedgerResultIT extends AbstractIT {
 
   @Test
   void getLedgerResult() throws JsonRpcClientErrorException {
-
-    final XrplClient mainnetClient = new MainnetEnvironment().getXrplClient();
     final LedgerResult ledgerResult = mainnetClient.ledger(LedgerRequestParams.builder()
       .ledgerSpecifier(LedgerSpecifier.VALIDATED)
       .transactions(true)
@@ -113,5 +117,28 @@ public class LedgerResultIT extends AbstractIT {
     );
     assertThat(ledgerResult.ledger().closeTimeHuman()).isNotEmpty();
     assertThat(ledgerResult.ledger().parentCloseTime()).isNotEmpty();
+  }
+
+  /**
+   * This test ensures that <a href="https://github.com/XRPLF/xrpl4j/issues/288">xrpl4j-228</a> has been fixed.
+   * This issue states that {@link LedgerResult}s which contain {@link org.xrpl.xrpl4j.model.transactions.UnlModify}
+   * transactions fail to get deserialized and therefore calls to {@link XrplClient#ledger(LedgerRequestParams)}
+   * fail. This test queries the ledgers found in the issue make sure xrpl4j can deserialize those ledgers.
+   */
+  @Test
+  void getLedgerResultWithTransactionsForLedgerWithUnlModify() throws JsonRpcClientErrorException {
+    LedgerResult ledgerResult = mainnetClient.ledger(
+      LedgerRequestParams.builder()
+        .ledgerSpecifier(LedgerSpecifier.of(73151744))
+        .transactions(true)
+        .build()
+    );
+
+    Optional<? extends Transaction> foundUnlModify = ledgerResult.ledger().transactions().stream()
+      .map(TransactionResult::transaction)
+      .filter(transaction -> UnlModify.class.isAssignableFrom(transaction.getClass()))
+      .findFirst();
+
+    assertThat(foundUnlModify).isNotEmpty().get().extracting("account").isEqualTo(UnlModify.ACCOUNT_ZERO);
   }
 }

--- a/xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/LedgerResultIT.java
+++ b/xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/LedgerResultIT.java
@@ -120,7 +120,7 @@ public class LedgerResultIT extends AbstractIT {
   }
 
   /**
-   * This test ensures that <a href="https://github.com/XRPLF/xrpl4j/issues/288">xrpl4j-228</a> has been fixed.
+   * This test ensures that <a href="https://github.com/XRPLF/xrpl4j/issues/288">xrpl4j-288</a> has been fixed.
    * This issue states that {@link LedgerResult}s which contain {@link org.xrpl.xrpl4j.model.transactions.UnlModify}
    * transactions fail to get deserialized and therefore calls to {@link XrplClient#ledger(LedgerRequestParams)}
    * fail. This test queries the ledgers found in the issue make sure xrpl4j can deserialize those ledgers.

--- a/xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/UnlModify.java
+++ b/xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/UnlModify.java
@@ -18,6 +18,8 @@ import org.xrpl.xrpl4j.model.client.common.LedgerIndex;
 @JsonDeserialize(as = ImmutableUnlModify.class)
 public interface UnlModify extends Transaction {
 
+  Address ACCOUNT_ZERO = Address.of("rrrrrrrrrrrrrrrrrrrrrhoLvTp");
+
   /**
    * Construct a builder for this class.
    *
@@ -27,6 +29,23 @@ public interface UnlModify extends Transaction {
     return ImmutableUnlModify.builder();
   }
 
+
+  /**
+   * This field is overridden in this class because of a bug in rippled that causes this field to be missing
+   * in API responses. In other pseudo-transactions such as {@link SetFee} and {@link EnableAmendment}, the rippled
+   * API sets the {@code account} field to a special XRPL address called ACCOUNT_ZERO, which is the base58
+   * encoding of the number zero. Because rippled does not set the {@code account} field of the {@link UnlModify}
+   * pseudo-transaction, this override will always set the field to ACCOUNT_ZERO to avoid deserialization issues
+   * and to be consistent with other pseudo-transactions.
+   *
+   * @return Always returns ACCOUNT_ZERO, which is the base58 encoding of the number zero.
+   */
+  @Override
+  @JsonProperty("Account")
+  @Value.Derived
+  default Address account() {
+    return ACCOUNT_ZERO;
+  }
 
   /**
    * The {@link org.xrpl.xrpl4j.model.client.common.LedgerIndex} where this pseudo-transaction appears.

--- a/xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/transactions/UnlModifyTest.java
+++ b/xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/transactions/UnlModifyTest.java
@@ -13,7 +13,6 @@ public class UnlModifyTest {
   public void testBuilder() {
     String validator = "EDB6FC8E803EE8EDC2793F1EC917B2EE41D35255618DEB91D3F9B1FC89B75D4539";
     UnlModify unlModify = UnlModify.builder()
-      .account(Address.of("rrrrrrrrrrrrrrrrrrrrrhoLvTp"))
       .fee(XrpCurrencyAmount.ofDrops(12))
       .sequence(UnsignedInteger.valueOf(2470665))
       .signingPublicKey("")
@@ -23,7 +22,7 @@ public class UnlModifyTest {
       .build();
 
     assertThat(unlModify.transactionType()).isEqualTo(TransactionType.UNL_MODIFY);
-    assertThat(unlModify.account()).isEqualTo(Address.of("rrrrrrrrrrrrrrrrrrrrrhoLvTp"));
+    assertThat(unlModify.account()).isEqualTo(UnlModify.ACCOUNT_ZERO);
     assertThat(unlModify.fee().value()).isEqualTo(UnsignedLong.valueOf(12));
     assertThat(unlModify.sequence()).isEqualTo(UnsignedInteger.valueOf(2470665));
     assertThat(unlModify.ledgerSequence()).isEqualTo(LedgerIndex.of(UnsignedInteger.valueOf(67850752)));

--- a/xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/transactions/json/UnlModifyJsonTests.java
+++ b/xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/transactions/json/UnlModifyJsonTests.java
@@ -15,7 +15,6 @@ public class UnlModifyJsonTests  extends AbstractJsonTest {
   @Test
   public void testJson() throws JsonProcessingException, JSONException {
     UnlModify unlModify = UnlModify.builder()
-      .account(Address.of("rrrrrrrrrrrrrrrrrrrrrhoLvTp"))
       .fee(XrpCurrencyAmount.ofDrops(12))
       .sequence(UnsignedInteger.valueOf(2470665))
       .signingPublicKey("")
@@ -25,7 +24,7 @@ public class UnlModifyJsonTests  extends AbstractJsonTest {
       .build();
 
     String json = "{" +
-      "\"Account\":\"rrrrrrrrrrrrrrrrrrrrrhoLvTp\"," +
+      "\"Account\":\"" + UnlModify.ACCOUNT_ZERO + "\"," +
       "\"Fee\":\"12\"," +
       "\"LedgerSequence\":67850752," +
       "\"Sequence\":2470665," +

